### PR TITLE
#159545736 Implement privacy defined restrictions for resource access

### PR DESCRIPTION
**Pull request by tomipaul**
#### What does this PR do?
Implements privacy defined restrictions for resource access

#### Description of completed tasks
- Check and apply resource privacy when a GET request is made to fetch a resource

#### How should this be manually tested?
- cd into project root directory and run `go test` command

#### Any background context?
Currently, an authenticated user can access any resource.

This PR aims to implement privacy by adding the following restrictions:
- When the privacy of a resource is **public**, **any authenticated user can access it**
- When the privacy of a resource is **private**, **only the owner of the resource can access it**
- When the privacy of a resource is **followers**, **it should be accessible to only the followers of the resource owner**

#### What are the relevant pivotal tracker stories?
#159545736

### CODE REVIEW

File: resource_test.go [hunk 1]
```diff
@@ -194,42 +227,143 @@ func TestGetResource(t *testing.T) {
 			Set("authorization", userToken).
```
Comment (mikey2020) [inline resource_test.go:227]: @tomipaul What is this test for?

**Comment by mikey2020** [comment]
@tomipaul on your deployed app 
`"ERROR #42P01 relation \"users\" does not exist (addr=\"10.237.182.66:5432\")"`

File: resource_test.go (hunk 1 above)
Comment (tomipaul) [reply resource_test.go:227]: @mikey2020 which of the tests? The one for invalid resource id or the one for nonexistent resource?

**Comment by tomipaul** [comment]
@mikey2020 Yes that is because the database have not been set up for the deployed app, I have to run the migrations to create the tables. I will do that now.

**Pull request merged**